wandb/sdk/internal/datastore.py

```diff
@@ -156,21 +177,28 @@ def _write_header(self):
             LEVELDBLOG_HEADER_MAGIC,
             LEVELDBLOG_HEADER_VERSION,
         )
-        assert len(data) == 7
+        assert (
+            len(data) == LEVELDBLOG_HEADER_LEN
+        ), "header size is {} bytes, expected {}".format(
+            len(data), LEVELDBLOG_HEADER_LEN
+        )
         self._fp.write(data)
         self._index += len(data)
 
     def _read_header(self):
-        header_length = 7
-        header = self._fp.read(header_length)
+        header = self._fp.read(LEVELDBLOG_HEADER_LEN)
+        assert (
+            len(header) == LEVELDBLOG_HEADER_LEN
+        ), "header is {} bytes instead of the expected {}".format(
+            len(header), LEVELDBLOG_HEADER_LEN
+        )
         ident, magic, version = struct.unpack("<4sHB", header)
         if ident != strtobytes(LEVELDBLOG_HEADER_IDENT):
             raise Exception("Invalid header")
         if magic != LEVELDBLOG_HEADER_MAGIC:
             raise Exception("Invalid header")
         if version != LEVELDBLOG_HEADER_VERSION:
             raise Exception("Invalid header")
-        assert len(header) == header_length
```

Comment:
This was one bug where we didn't verify we had a valid header before attempt to unpack it.